src/import-data/ImportData.ts

```diff
@@ -1,72 +1,88 @@
 import _ from 'lodash/fp';
-import { Fragment, useCallback, useState } from 'react';
-import { h } from 'react-hyperscript-helpers';
+import { Fragment, ReactNode, useState } from 'react';
+import { div, h } from 'react-hyperscript-helpers';
 import { spinnerOverlay } from 'src/components/common';
 import { Ajax } from 'src/libs/ajax';
 import { Dataset } from 'src/libs/ajax/Catalog';
 import { resolveWdsUrl, WdsDataTableProvider } from 'src/libs/ajax/data-table-providers/WdsDataTableProvider';
+import colors from 'src/libs/colors';
 import { withErrorReporting } from 'src/libs/error';
 import Events, { extractWorkspaceDetails } from 'src/libs/events';
 import * as Nav from 'src/libs/nav';
 import { notify } from 'src/libs/notifications';
 import { useOnMount } from 'src/libs/react-utils';
 import { asyncImportJobStore } from 'src/libs/state';
 import * as Utils from 'src/libs/utils';
+import { WorkspaceInfo } from 'src/libs/workspace-utils';
 import { useDataCatalog } from 'src/pages/library/dataBrowser-utils';
 import { notifyDataImportProgress } from 'src/workspace-data/import-jobs';
 
-import { TemplateWorkspaceInfo } from './import-types';
+import {
+  BagItImportRequest,
+  CatalogDatasetImportRequest,
+  CatalogSnapshotsImportRequest,
+  EntitiesImportRequest,
+  ImportRequest,
+  PFBImportRequest,
+  TDRSnapshotExportImportRequest,
+  TDRSnapshotReferenceImportRequest,
+  TemplateWorkspaceInfo,
+} from './import-types';
 import { ImportDataDestination } from './ImportDataDestination';
 import { ImportDataOverview } from './ImportDataOverview';
 import { isProtectedSource } from './protected-data-utils';
+import { useImportRequest } from './useImportRequest';
 
-// ImportData handles all the information relating to the page itself - this includes:
-// * Reading from the URL
-// * Loading initial Data
-// * Managing the import
-export const ImportData = () => {
+const getTitleForImportRequest = (importRequest: ImportRequest): string => {
+  if (importRequest.type === 'tdr-snapshot-export') {
+    return `Importing snapshot ${importRequest.snapshotName}`;
+  }
+
+  if (
+    importRequest.type === 'tdr-snapshot-reference' ||
+    importRequest.type === 'catalog-dataset' ||
+    importRequest.type === 'catalog-snapshots'
```

Comment:
I'm sad that our `switchCase` doesn't allow multiple matching values.

Reply:
Alternatively, this could be something like `if (['tdr-snapshot-reference', ...].includes(importRequest.type)) {`, but writing it this way lets TypeScript narrow the type of `importRequest` and made it easier for me to verify that only the cases I expect have fallen through to the default return value.

Reply:
Can't `switch` be used here? TS can infer types from `switch` just like it can from `if`.
```
  switch (importRequest.type) {
    case 'tdr-snapshot-export':
      return `Importing snapshot ${importRequest.snapshotName}`;
    case 'tdr-snapshot-reference':
    case 'catalog-dataset':
    case 'catalog-snapshots':
      return 'Linking data to a workspace';
    default:
      return 'Importing data to a workspace';
  }
```

Reply:
Replaced with `switch` in https://github.com/DataBiosphere/terra-ui/pull/4300/commits/0dbe987e2eaa2400b6befbafd7ef9985c90412b3.